From 537522cd68a2fbd0dce481f8e0389b128604765e Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Mon, 29 Apr 2013 11:06:04 +0200
Subject: [PORT FROM MAIN] crashtool: correct a few bugs, simplify implementation

BZ: 104530

remove unused context structure, add missing define(s).
also add missing includes.

Change-Id: I947f3c526ee8a9ebbedd9b5721d3c3913b1a2e07
Orig-Change-Id: I564498bfa14272a705c599786811df6f8b679bbe
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/105314
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libc/include/ctm.h |   22 ++++++++++------------
 libc/include/lct.h |    6 ++++++
 2 files changed, 16 insertions(+), 12 deletions(-)

diff --git a/libc/include/ctm.h b/libc/include/ctm.h
index a45611b..025ffb5 100644
--- a/libc/include/ctm.h
+++ b/libc/include/ctm.h
@@ -18,21 +18,21 @@
 
 #  define max(X, Y)  ((X) < (Y) ? (Y) : (X))
 
-#  define STAT_TRIG_SUFFIX  "_trigger"
 #  define CRASH_TRIG_SUFFIX "_trigger"
 #  define INFO_TRIG_SUFFIX  "_infoevent"
+#  define STAT_TRIG_SUFFIX  "_trigger"
 
-#  define INFO_DATA_SUFFIX "_data"
 #  define CRASH_DATA_SUFFIX "_data"
+#  define INFO_DATA_SUFFIX "_data"
+#  define STAT_DATA_SUFFIX "_data"
 
-struct ct_monitor_ctx {
-	int thread_should_stop;
-	int sock_nl_fd;
-	int sock_fd;
-	char *path_stat_dir;
-	char *path_crash_dir;
-	char *path_info_dir;
-};
+#  define CRASH_PATH_DIR "/logs/cwscrash"
+#  define INFO_PATH_DIR "/logs/stats"
+#  define STAT_PATH_DIR "/logs/stats"
+
+extern int thread_should_stop;
+extern int sock_nl_fd;
+extern int sock_fd;
 
 enum ctm_ev_pending {
 	EV_PENDING_ALL,
@@ -45,10 +45,8 @@ typedef int (*ev_handler_t)(struct ct_event *ev);
 
 /* ctm_netlink.c */
 extern int ctm_nl_init(void);
-extern int ctm_nl_exit(int fd);
 extern int ctm_nl_sendto_kct(int fd, int type, const void *data, unsigned int size);
 extern struct kct_packet *ctm_nl_get_packet(int fd);
-extern int ctm_nl_dclr_pid(int fd);
 
 /* ctm_comm.c */
 extern enum ctm_ev_pending	ctm_comm_wait_event(long timeout);
diff --git a/libc/include/lct.h b/libc/include/lct.h
index 0748304..190a79d 100644
--- a/libc/include/lct.h
+++ b/libc/include/lct.h
@@ -1,6 +1,12 @@
 #ifndef LCT_H_
 # define LCT_H_
 
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <linux/kct.h>
-- 
1.7.4.1

